_posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc

```diff
@@ -0,0 +1,248 @@
+
+The above subproject provides the JBeret module that can be used in the WildFly server. It uses the concept of WildFly Channel to achieve this goal. It can be used to override the version of some artifacts used in modules inside the WildFly server. I won’t explain the concept of WildFly Channel in this blog post. If you are interested in this topic, these articles are worth reading:
+
+* https://www.wildfly.org/news/2023/04/05/prospero/[An Introduction To The Prospero Usage]
+* https://www.wildfly.org/news/2024/01/29/wildfly-glow/[WildFly Glow, an evolution of WildFly provisioning]
+* https://www.wildfly.org/news/2024/01/31/whats-new-in-provisioning/[What’s new in WildFly provisioning]
+
+With the above configuration, the `3.0.0.Final` version of that manifest specifies use of JBeret `3.0.0` while WildFly `33.0.2` used `2.2.1`. If I run the following command in the above example project, the output showing the relative provision process:
+
+[source,bash]
+----
+$ mvn wildfly:run
+...
+[INFO] Resolving channel metadata from Maven artifact org.jberet:jberet-channel-manifest:3.0.0.Final
+[INFO] Provisioning server in /Users/weli/works/jberet-examples/deployment/target/wildfly
+[INFO] Resolving feature-packs
+[INFO] Installing packages
+[INFO] 6 of 523 (1.1%)
+[INFO] Resolving artifacts
+[INFO] 11 of 491 (2.2%)
+...
+[INFO] Generating configurations
+[INFO] Delayed generation, waiting...
+[INFO] Resolving channel metadata from Maven artifact org.jberet:jberet-channel-manifest:3.0.0.Final
+...
+[INFO] --- wildfly:5.0.1.Final:start (wildfly-start) @ batch-deployment-examples ---
+[INFO] Provisioning default server in /Users/weli/works/jberet-examples/deployment/target/server
+Downloading from jboss-public-repository-group: https://repository.jboss.org/nexus/content/groups/public/org/wildfly/wildfly-galleon-pack/33.0.2.Final/wildfly-galleon-pack-33.0.2.Final.zip
+...
+----
+
+From the above output, we can see that `org.jberet:jberet-channel-manifest:3.0.0.Final` is used for provision, and the `wildfly-galleon-pack-33.0.2.Final.zip` is downloaded. If I check the generated WildFly server directory, I can see that JBeret `3.0.0.Final` is used in the provisioned WildFly server:
+
+[source,bash]
+----
+$ pwd
+/Users/weli/works/jberet-examples/deployment/target/wildfly
+$ find modules -name "*jberet*"
+modules/system/layers/base/org/wildfly/extension/batch/jberet
+modules/system/layers/base/org/wildfly/extension/batch/jberet/main/wildfly-batch-jberet-33.0.2.Final.jar
+modules/system/layers/base/org/jberet
+modules/system/layers/base/org/jberet/jberet-core
+modules/system/layers/base/org/jberet/jberet-core/main/jberet-core-3.0.0.Final.jar
+----
+
+To run the provisioned WildFly server in debug mode, the `wildfly-maven-plugin` provide options to support the debug mode. To check the usages of `wildfly-maven-plugin`, I can run the following command in the above example project to do so:
```

Comment:
You could also link to the documentation where https://docs.wildfly.org/wildfly-maven-plugin.

Reply:
Okay! I have added a `Note` section in below to provide the additional information.